components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,90 +486,360 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
-                    if (SCIMCommonConstants.EQ.equalsIgnoreCase(filterOperation)) {
-                        if (StringUtils.equals(attributeName, SCIMConstants.UserSchemaConstants.USER_NAME_URI) &&
-                                !StringUtils.contains(attributeValue, CarbonConstants.DOMAIN_SEPARATOR)) {
-                            attributeValue = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME +
-                                    CarbonConstants.DOMAIN_SEPARATOR + attributeValue;
-                        }
-                    }
+            // 0th index is to store total number of results.
+            filteredUsers.add(0);
+
+            // Support multi attribute filtering.
+            return getMultiAttributeFilteredUsers(node, requiredAttributes, offset, limit, sortBy, sortOrder,
+                    domainName, filteredUsers);
+        } else {
+            throw new CharonException("Unknown operation. Not either an expression node or an operation node.");
+        }
+    }
+
+    /**
+     * Method to filter users for a filter with a single attribute.
+     *
+     * @param node               Expression node for single attribute filtering
+     * @param requiredAttributes Required attributes for the response
+     * @param offset             Starting index of the count
+     * @param limit              Counting value
+     * @param sortBy             SortBy
+     * @param sortOrder          Sorting order
+     * @param domainName         Domain to run the filter
+     * @return User list with detailed attributes
+     * @throws CharonException Error while filtering
+     */
+    private List<Object> getFilteredUsersBySingleAttribute(ExpressionNode node, Map<String, Boolean> requiredAttributes,
+            int offset, int limit, String sortBy, String sortOrder, String domainName) throws CharonException {
+
+        String[] userNames;
+
+        // Set filter values.
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
+
+        if (log.isDebugEnabled()) {
+            log.debug(
+                    String.format("Listing users by filter: %s %s %s", attributeName, filterOperation, attributeValue));
+        }
+        try {
+
+            validateFilterOperation(filterOperation);
+
+            // Apply filter enhancements for username and update expression node value.
+            attributeValue = applyFilterEnhancementsForUsername(filterOperation, attributeName, attributeValue, node);
+
+            if ((carbonUM instanceof PaginatedUserStoreManager) && isPaginatedUserStoreAvailable()) {
+
+                // Set the limit if specified.
+                if (limit <= 0)
+                    limit = getMaxLimit();
+
+                // Extract the domain name if the domain name is embedded in the filter attribute value.
+                domainName = resolveDomainInAttributeValueAndUpdateExpressionNode(domainName, node);
```

Comment:
Shall we change the private method name resolveDomainInAttributeValueAndUpdateExpressionNode? Usually we do a single task from a single method, hence it is a good practice to avoid using 'And' in a method name

Reply:
fix - ac4cf58d8f20ee81c08cb55ec47eed9ce3c305ed